packages/shared/src/utils/currency/currency.ts

```diff
@@ -65,53 +93,45 @@ export const formatMoney = (currency_value, amount, exclude_currency, decimals =
     return sign + (exclude_currency ? '' : formatCurrency(currency_value)) + money;
 };
 
-export const formatCurrency = currency => {
+export const formatCurrency = (currency: string) => {
     return `<span class="symbols ${(currency || '').toLowerCase()}"></span>`;
 };
 
-export const addComma = (num, decimal_points, is_crypto) => {
-    let number = String(num || 0).replace(/,/g, '');
+export const addComma = (num?: number | string | null, decimal_points?: number, is_crypto?: boolean) => {
+    let number: number | string = String(num || 0).replace(/,/g, '');
     if (typeof decimal_points !== 'undefined') {
         number = (+number).toFixed(decimal_points);
     }
     if (is_crypto) {
-        number = parseFloat(+number);
+        number = parseFloat(String(number));
     }
 
     return number
         .toString()
         .replace(/(^|[^\w.])(\d{4,})/g, ($0, $1, $2) => $1 + $2.replace(/\d(?=(?:\d\d\d)+(?!\d))/g, '$&,'));
 };
 
-export const calcDecimalPlaces = currency => {
+export const calcDecimalPlaces = (currency: string) => {
     return isCryptocurrency(currency) ? getPropertyValue(CryptoConfig.get(), [currency, 'fractional_digits']) : 2;
 };
 
-export const getDecimalPlaces = currency =>
+export const getDecimalPlaces = (currency: string) =>
     // need to check currencies_config[currency] exists instead of || in case of 0 value
     currencies_config[currency]
         ? getPropertyValue(currencies_config, [currency, 'fractional_digits'])
         : calcDecimalPlaces(currency);
 
-export const hasCorrectDecimalPlaces = (currency, amount) => {
-    const currency_decimal_places = getDecimalPlaces(currency);
-    const amount_decimal_array = amount.toString().split('.')[1];
-    const amount_decimal_places = amount_decimal_array ? amount_decimal_array.length || 0 : 0;
-
-    return amount_decimal_places <= currency_decimal_places;
-};
-
-export const setCurrencies = website_status => {
+export const setCurrencies = (website_status: { currencies_config: TCurrenciesConfig }) => {
     currencies_config = website_status.currencies_config;
 };
 
 // (currency in crypto_config) is a back-up in case website_status doesn't include the currency config, in some cases where it's disabled
-export const isCryptocurrency = currency => {
+export const isCryptocurrency = (currency: string) => {
     return /crypto/i.test(getPropertyValue(currencies_config, [currency, 'type'])) || currency in CryptoConfig.get();
 };
 
 export const CryptoConfig = (() => {
-    let crypto_config;
+    let crypto_config: any;
```

Comment:
The returntype fo `initCryptoConfig` is any, so I added any type to this!

Reply:
@niloo-fs i have to disagree because it has an exact readonly type which is the following:
```suggestion
    let crypto_config: TCryptoConfig;
```
please declare a type for it on the top of the file:
```
type TCryptoConfig = Readonly<{
    [key: string]: {
        display_code: string;
        name: string;
        min_withdrawal: number;
        pa_max_withdrawal: number;
        pa_min_withdrawal: number;
        fractional_digits: number;
    };
}>;
```
it's type is the type of a deeply frozen object returned from CryptoConfig.get(); 🙂